synapse/http/federation/well_known_resolver.py

```diff
@@ -130,9 +128,10 @@ async def get_well_known(self, server_name: bytes) -> WellKnownLookupResult:
         # requests for the same server in parallel?
         try:
             with Measure(self._clock, "get_well_known"):
-                result, cache_period = await self._fetch_well_known(
-                    server_name
-                )  # type: Optional[bytes], float
+                result: Optional[bytes]
+                cache_period: float
+
+                result, cache_period = await self._fetch_well_known(server_name)
```

Comment:
I find this fairly unreadable and would prefer the old way, but 🤷

Reply:
Some SO answers say this is the way to do it, i think this is the most pythonic you're gonna get, an alternative would be
```py
res: Tuple[Optional[bytes], float] = await self._fetch_well_known(server_name)

result: Optional[bytes] = res[0]
cache_period: float = res[1]
```

Reply:
In my opinion, neither before or after is particularly pretty, but I guess there's something to be said about adhering to the 'Pythonic'/standard way of doing things, and if this is it, then why not?

Reply:
Yeah I can't say the alternatives are very pretty either. I don't think we'd even need them except we have to mark `result` as optional. 😢

Reply:
I think that explicitly annotating unpacked variables wont be needed if we have enough function signatures in place, after which we can just drop the inline type arguments (unless its needed for some explicit reason) and let mypy infer from the methods.

So i think this is okay coupled with the reason to please mypy for now, with the intention to remove it later, does that sound good?

Reply:
> I think that explicitly annotating unpacked variables wont be needed if we have enough function signatures in place

`_fetch_well_known` *does* have proper types, it just that it is `Tuple[bytes, float]`, but the caller might have the `result` value be `Optional[bytes]` so it gets a bit awkward. 😄 

Anyway, I think this is fine.